src/bot.rs

```diff
@@ -0,0 +1,44 @@
+use engine::GameState;
+
+pub trait Bot {
+    fn tick(&mut self, game_state: &GameState) -> ActionSet;
+    fn game_start(&mut self) {}
+    fn game_end(&mut self) {}
+}
+
+impl<F> Bot for F
+where
+    F: FnMut(&GameState) -> ActionSet,
```

Comment:
I'm a little bit confused by these few lines. Is this saying a the user of the lib can implement a there own definition of the the `tick()` function does for their own bot? Where `tick()` is a mutable function that accepts a reference to `GateState` and must return an `ActionSet`?

Reply:
Yep, it means that you can use a closure as a bot, like I did in `examples/simple_bot.rs`. It's mainly for convenience or simple bots where you just want to do something every tick and don't care about the start and end lifecycle methods.